drake/systems/plants/@RigidBodyManipulator/doKinematics.m

```diff
@@ -83,7 +84,17 @@
 kinsol.v = v;
 
 if (options.use_mex && model.mex_model_ptr~=0 && isnumeric(q))
-  doKinematicsmex(model.mex_model_ptr,q,options.compute_gradients,v,options.compute_JdotV);
+  if isempty(options.kinematics_cache_ptr_to_use)
+    if options.compute_gradients
+      kinsol.mex_ptr = model.default_kinematics_cache_ptr_with_gradients;
```

Comment:
doesn't this break the contract?  If you're always overwriting the same existing kinematics cache object, then if I call 
```
kinsol = doKinematics(q1)
kinsol2 = doKinematics(q2)
```
are they going to be pointing to the same cache?